src/components/molecules/Visualizer/Engine/Cesium/Box/index.tsx

```diff
@@ -20,75 +35,342 @@ export type Property = {
     outlineColor?: string;
+    activeOutlineColor?: string;
     outlineWidth?: number;
-    fill?: boolean;
-    outline?: boolean;
+    draggableOutlineColor?: string;
+    activeDraggableOutlineColor?: string;
+    draggableOutlineWidth?: number;
+    scalePoint?: boolean;
+    axisLine?: boolean;
+    pointFillColor?: string;
+    pointOutlineColor?: string;
+    activePointOutlineColor?: string;
+    pointOutlineWidth?: number;
+    axisLineColor?: string;
+    axisLineWidth?: number;
+    keepBoxAboveGround?: boolean;
+    cursor?: string;
+    activeBox?: boolean;
+    activeScalePointIndex?: number; // 0 ~ 11
+    activeEdgeIndex?: number; // 0 ~ 11
   };
 };
 
-// The 6 box sides defined as planes in local coordinate space.
-// ref: https://github.com/TerriaJS/terriajs/blob/cad62a45cbee98c7561625458bec3a48510f6cbc/lib/Models/BoxDrawing.ts#L161-L169
-const SIDE_PLANES: readonly CesiumPlane[] = [
-  new CesiumPlane(new Cartesian3(0, 0, 1), 0.5),
-  new CesiumPlane(new Cartesian3(0, 0, -1), 0.5),
-  new CesiumPlane(new Cartesian3(0, 1, 0), 0.5),
-  new CesiumPlane(new Cartesian3(0, -1, 0), 0.5),
-  new CesiumPlane(new Cartesian3(1, 0, 0), 0.5),
-  new CesiumPlane(new Cartesian3(-1, 0, 0), 0.5),
-];
-
-// This is used for plane ID.
-// We can use this name, for example you want to move the box when front plane is dragged.
-const SIDE_PLANE_NAMES = ["bottom", "top", "front", "back", "left", "right"];
-
-const updateTrs = (trs: TranslationRotationScale, property: Property | undefined) => {
-  const { location, height, width, length } = property?.default ?? {};
-
-  const translation = location
-    ? Cartesian3.fromDegrees(location.lng, location.lat, location.height ?? 0)
-    : undefined;
-  if (translation) {
-    Cartesian3.clone(translation, trs.translation);
-  }
-
-  // Quaternion.clone(trs.rotation, this.trs.rotation);
+const Box: React.FC<PrimitiveProps<Property>> = memo(function BoxPresenter({ layer }) {
+  const { viewer } = useCesium();
+  const ctx = useContext();
+  const { id, isVisible, property } = layer ?? {};
+  const {
+    fillColor,
+    outlineColor,
+    activeOutlineColor,
+    outlineWidth,
+    draggableOutlineColor,
+    activeDraggableOutlineColor,
+    draggableOutlineWidth,
+    height = 100,
+    width = 100,
+    length = 100,
+    pointFillColor,
+    pointOutlineColor,
+    activePointOutlineColor,
+    pointOutlineWidth,
+    axisLineColor,
+    axisLineWidth,
+    keepBoxAboveGround,
+    scalePoint = true,
+    axisLine = true,
+    cursor,
+    activeBox,
+    activeEdgeIndex,
+    activeScalePointIndex,
+  } = property?.default ?? {};
 
-  Cartesian3.clone(new Cartesian3(width || 100, length || 100, height || 100), trs.scale);
+  const [terrainHeightEstimate, setTerrainHeightEstimate] = useState(0);
+  const [trs] = useState(() =>
+    updateTrs(new TranslationRotationScale(), property, terrainHeightEstimate),
+  );
 
-  return trs;
-};
+  const isProgressSamplingTerrainHeight = useRef(false);
```

Comment:
Is this suppose to be `inProgressSamplingTerrainHeight`? This is just keeping track of whether we are calculating the height, right?

Reply:
I renamed this in [refactor: rename isProgressSamplingTerrainHeight](https://github.com/reearth/reearth-web/pull/357/commits/8b4ce98f4db663409909e18faa2ced0d7a418d56)